README.md

```diff
@@ -1 +1,46 @@
 # rancher-desktop-networking
+Is a simple layer 2 switch that allows connectivity between the VM and the host over `AF_VSOCK` written in Go using [gvisor's](https://github.com/google/gvisor) network stack and [gvisor-tap-vsock](https://github.com/google/gvisor). In addition to piping the ethernet frames, it also provides a configurable DNS, dynamic port forwarding and DHCP.
+
+Rancher Desktop Networking consists of three main components: host-switch, vm-switch, and network-setup. Below is a brief explanation of what each component does.
+
+## host-switch
+For now, it only runs on Windows and acts as a receiving end for all traffic that originated from the WSL VM (designated namespace). It performs a handshake to find the right VM to talk to over `AF_VSOCK`. Once a correct VM is found, it then listens for the incoming traffic from that VM. In additon to this, it can provide a configurable DNS resolver that runs in the user space network along with an API that allows for dynamic port forwarding.
+
+## network-setup
+Its main responsibility is to create a network namespace and start the `vm-switch` subprocess in the newly created network namespace.
```

Comment:
… running `init` is kind of important, too?

Reply:
I don't know if init belongs in this documentation, that's more of the Rancher Desktop operation domain. But, if you have any suggestions I'm happy to include them.

Reply:
`network-setup` still takes an arg for stuff to run under unshare, so we should at least document that part.